src/containers/repo.jsx

```diff
@@ -1969,15 +1998,28 @@ class Repo extends Component {
        <ClipLoader animation="border"
           size={12} color={"#FFFFFF"}
           loading={this.state.publishing}/>
-    </button>
+      </button>
     );
     if (this.props.env === 'local') {
       // Hide buttons which are not implemented in local environment yet
       // TODO - Implement rendering in backend and show previewButton 
       previewButton = null;
       shareButton = null;
       publishButton = null;
-      if (this.state.saved) {
+      if (group != undefined) {
+        publishButton = (
```

Comment:
I'd rather not duplicate the publishButton code, but have just one button and then branch out the logic in this.publish() method (based on whether we're running the local or the cloud version of the UI)

Reply:
How should we handle the publish button when the group is undefined?

I'm not sure how Minerva Cloud handles this, but for the local version it only makes sense to publish if there is at least one channel group, so I would favor only showing the publish button if there is an active channel group.

Reply:
I've just set the publish button to only appear if there is an active group.

Reply:
I agree, that's a sensible approach 👍